TeachingRecordSystem/src/TeachingRecordSystem.Core/DataStore/Postgres/Models/Person.cs

```diff
@@ -446,6 +442,29 @@ public bool RefreshProfessionalStatusAttributes(
         }
     }
 
+    public bool RefreshProfessionalStatusAttributesForUpdate(
+        ProfessionalStatusType professionalStatusType,
+        IReadOnlyCollection<RouteToProfessionalStatus> allRoutes)
+    {
+        var professionalStatuses =
+            Qualifications?.OfType<ProfessionalStatus>() ??
+            throw new InvalidOperationException("No professional statuses.");
+
+        return RefreshProfessionalStatusAttributes(professionalStatusType, allRoutes, professionalStatuses);
+    }
+
+    public bool RefreshProfessionalStatusAttributesForDelete(
```

Comment:
If we're splitting these out and having different methods for each operation then we should have `RefreshProfessionalStatusAttributesForCreate` as well.

Reply:
removed